File: front-end/app/models/category.js
// // import DS from 'ember-data';

// // export default DS.Model.extend({

// // });
// import Ember from 'ember';

// export default Ember.Object.extend({
    
// });
import DS from 'ember-data';

export default DS.Model.extend({
    'name': DS.attr('string'),
    'description': DS.attr('string'),
    'uuid': DS.attr('string'),
    'products': DS.hasMany('product')
});

File: jft-market/src/main/java/com/jft/market/model/Category.java
package com.jft.market.model;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "categories")
@NoArgsConstructor
@Getter
@Setter
public class Category extends TimestampedFieldObject {

	@javax.persistence.Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long Id;
	@Column(unique = true)
	private String name;
	private String description;
	private String uuid;

	@ManyToMany(mappedBy = "categories")
	private Set<Product> products = new HashSet<>();

	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "parent_category_id")
	private Category category;

	@OneToMany(mappedBy = "category")
	private Set<Category> subCategories = new HashSet<>();
}
